docs/content/latest/api/ysql/extensions.md

```diff
@@ -0,0 +1,361 @@
+
+Connect with `ysqlsh` and run the commands below.
+```sql
+CREATE EXTENSION insert_username;
+CREATE EXTENSION moddatetime;
+
+CREATE TABLE spi_test (
+  id int primary key,
+  content text,
+  username text not null,
+  moddate timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+CREATE TRIGGER insert_usernames
+  BEFORE INSERT OR UPDATE ON spi_test
+  FOR EACH ROW
+  EXECUTE PROCEDURE insert_username (username);
+
+CREATE TRIGGER update_moddatetime
+  BEFORE UPDATE ON spi_test
+  FOR EACH ROW
+  EXECUTE PROCEDURE moddatetime (moddate);
+```
+
+- Insert some rows
+
+Each insert should add the current role as `username` and the current timestamp as `moddate`.
+
+{{< note title="Note" >}}
+
+YSQL should have users `yugabyte` and (for compatibility) `postgres` created by default. 
+
+{{< /note >}}
+
+```sql
+SET ROLE yugabyte;
+INSERT INTO spi_test VALUES(1, 'desc1');
+
+SET ROLE postgres;
+INSERT INTO spi_test VALUES(2, 'desc2');
+INSERT INTO spi_test VALUES(3, 'desc3');
+
+SET ROLE yugabyte;
+INSERT INTO spi_test VALUES(4, 'desc4');
+
+SELECT * FROM spi_test ORDER BY id;
+```
+
+```
+ id | content | username |          moddate
+----+---------+----------+----------------------------
+  1 | desc1   | yugabyte | 2019-09-13 16:55:53.969907
+  2 | desc2   | postgres | 2019-09-13 16:55:53.983306
+  3 | desc3   | postgres | 2019-09-13 16:55:53.98658
+  4 | desc4   | yugabyte | 2019-09-13 16:55:53.991315
+(4 rows)
+```
+
+- Update some rows.
+
+Should update both `username`  and `moddate` accordingly.
+
+```sql
+UPDATE spi_test SET content = 'desc1_updated' WHERE id = 1;
+UPDATE spi_test SET content = 'desc3_updated' WHERE id = 3;
+
+SELECT * FROM spi_test ORDER BY id;
+```
+
+```
+ id |    content    | username |          moddate
+----+---------------+----------+----------------------------
+  1 | desc1_updated | yugabyte | 2019-09-13 16:56:27.623513
+  2 | desc2         | postgres | 2019-09-13 16:55:53.983306
+  3 | desc3_updated | yugabyte | 2019-09-13 16:56:27.634099
+  4 | desc4         | yugabyte | 2019-09-13 16:55:53.991315
+(4 rows)
+```
+
+Read more about the spi module in the [PostgreSQL Docs](https://www.postgresql.org/docs/current/contrib-spi.html).
+
+
+## Install Other Extensions
+
+Other extensions have to be installed manually into the YugaByte `lib` folder before they can be enabled (with `CREATE EXTENSION`).
+
+{{< note title="Note" >}}
+
+Currently, in a multi-node setup, the installation instructions below must be done on _every_ node in the cluster.
+
+{{< /note >}}
+
+
+Typically extensions need three types of files:
+
+- Shared library files (`<name>.so`)
+- SQL files (`<name>--<version>.sql`)
+- Control files (`<name>.control`)
+
+In order to install an extension you need to copy these files into the respective folders of your YugaByte installation.
+
+
+Shared library files will be in the `pkglibdir` directory while SQL and control files should be in the `extension` subdirectory of the 
+`libdir` directory.
```

Comment:
Arbitrary linebreak.  Is your character limit set to 140 or sth?  I think Yugabyte uses 100 as a standard.